test/spec/mfa-challenge.js

```diff
@@ -149,6 +344,304 @@ define(function(require) {
+                }
+              },
+              response: 'success'
+            }
+          ]
+        },
+        execute: function (test) {
+          return test.trans.poll({
+            delay: 0,
+            autoPush: true
+          });
+        }
+      });
+
+      util.itMakesCorrectRequestResponse({
+        title: 'allows polling for push with autoPush false',
+        setup: {
+          status: 'mfa-challenge-push',
+          calls: [
+            {
+              request: {
+                uri: '/api/v1/authn/factors/opf492vmb3s1blLTs0h7/verify?autoPush=false',
+                data: {
+                  stateToken: '00T4jcVNRzJy5dkWJ4P7c9051dY3FUYY9O2zvbU_vI'
+                }
+              },
+              response: 'mfa-challenge-push'
+            },
+            {
+              request: {
+                uri: '/api/v1/authn/factors/opf492vmb3s1blLTs0h7/verify?autoPush=false',
+                data: {
+                  stateToken: '00T4jcVNRzJy5dkWJ4P7c9051dY3FUYY9O2zvbU_vI'
+                }
+              },
+              response: 'mfa-challenge-push'
+            },
+            {
+              request: {
+                uri: '/api/v1/authn/factors/opf492vmb3s1blLTs0h7/verify?autoPush=false',
+                data: {
+                  stateToken: '00T4jcVNRzJy5dkWJ4P7c9051dY3FUYY9O2zvbU_vI'
+                }
+              },
+              response: 'success'
+            }
+          ]
+        },
+        execute: function (test) {
+          return test.trans.poll({
+            delay: 0,
+            autoPush: false
+          });
+        }
+      });
+
+      util.itMakesCorrectRequestResponse({
+        title: 'allows polling for push with autoPush as a function',
+        setup: {
+          status: 'mfa-challenge-push',
+          calls: [
+            {
+              request: {
+                uri: '/api/v1/authn/factors/opf492vmb3s1blLTs0h7/verify?autoPush=true',
+                data: {
+                  stateToken: '00T4jcVNRzJy5dkWJ4P7c9051dY3FUYY9O2zvbU_vI'
+                }
+              },
+              response: 'mfa-challenge-push'
+            },
+            {
+              request: {
+                uri: '/api/v1/authn/factors/opf492vmb3s1blLTs0h7/verify?autoPush=true',
+                data: {
+                  stateToken: '00T4jcVNRzJy5dkWJ4P7c9051dY3FUYY9O2zvbU_vI'
+                }
+              },
+              response: 'mfa-challenge-push'
+            },
+            {
+              request: {
+                uri: '/api/v1/authn/factors/opf492vmb3s1blLTs0h7/verify?autoPush=true',
+                data: {
+                  stateToken: '00T4jcVNRzJy5dkWJ4P7c9051dY3FUYY9O2zvbU_vI'
+                }
+              },
+              response: 'success'
+            }
+          ]
+        },
+        execute: function (test) {
+          return test.trans.poll({
+            delay: 0,
+            autoPush: function () {
+              return true;
```

Comment:
might be a good idea to return false in the third response just to test that it actually sends something different if the value is changed after the first poll request.

Reply:
I second this. We should change one of the polling tests to change between requests

Reply:
Done!